File: Administration System/Administration.Model/Types/HintUserView.cs

namespace Model.Types
{
    [ViewModel(typeof(HintUserView_Functions))]
    public class HintUserView
    {
        public HintUserView(int taskId, int hintNo, string title, string contents, int previousHintNo, int nextHintNo, int costOfNextHint)
        {
            TaskId = taskId;
            HintNo = hintNo;
            Title = title;
            Contents = contents;
            PreviousHintNo = previousHintNo;
            NextHintNo = nextHintNo;
            CostOfNextHint = costOfNextHint;
        }

        public int TaskId { get; init; }

        public int HintNo { get; init; }

        public string Title { get; init; }

        public string Contents { get; init; }

        public int PreviousHintNo { get; init; } //0 indicates there is no previous hint

        public int NextHintNo { get; init; } //0 indicates there is no next hint

        public int CostOfNextHint { get; init; }  //If zero means that the user can just navigate to it (because they have seen it before)
    }
}

File: Client/src/app/models/hint-user-view.ts
export interface IHintUserView {
    HintNo: number;
    Title: string;
    Contents: string;
    PreviousHintNo: number;//0 indicates there is no previous hint
    NextHintNo: number;//0 indicates there is no next hint
    NextHintAlreadyUsed: boolean;
}

export class HintUserView implements IHintUserView {
    HintNo = 0;
    Title = '';
    Contents = '';
    PreviousHintNo = 0;
    NextHintNo = 0;
    NextHintAlreadyUsed = false;
}

export const EmptyHintUserView = new HintUserView();